var/spack/repos/builtin/packages/chatterbug/package.py

```diff
@@ -0,0 +1,89 @@
+##############################################################################
+
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Chatterbug(MakefilePackage):
+    """A suite of communication-intensive proxy applications that mimic
+       commonly found communication patterns in HPC codes. These codes can be
+       used as synthetic codes for benchmarking, or for trace generation using
+       OTF2.
+    """
+    tags = ['proxy-app']
+
+    homepage = "https://chatterbug.readthedocs.io"
+    git      = "https://github.com/LLNL/chatterbug.git"
+
+    version('develop', branch='master')
+    version('1.0', tag='v1.0')
+
+    variant('otf2', default=False, description='Build with OTF2 tracing')
+
+    depends_on('mpi')
+    depends_on('scorep~gui', when='+otf2')
+
+    @property
+    def build_targets(self):
+        targets = []
+
+        cxxflag = ' -O3 '
+
+        if '+otf2' in self.spec:
+            cxxflag += '-I{0}'.format(spec['scorep'].prefix.include)
+            cxxflag += '-I{0}/scorep'.format(spec['scorep'].prefix.include)
```

Comment:
@bhatele is it only `Chatterbug` which needs `-I prefix.include.scorep`? 
Generally, I think it's the right design to have all headers to be in a separate folder so that downstream you do, for example, `#include <scrorep/utilities.h>` instead of `#include <utilities.h>`. The latter is bound to fail as soon as a common filename like `utilities.h` exists in multiple dependencies.

If you don't need `spec.prefix.include.scorep`, the compiler wrappers will work OOB and you don't need to bother with include flags at all.